marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/validateMergingStep.sjs

```diff
@@ -17,43 +17,44 @@
 
 const validateMergeLib = require("/data-hub/5/data-services/mastering/validateMergingStepLib.sjs");
 
-var stepName;
+var stepName, view;
 
 xdmp.securityAssert("http://marklogic.com/data-hub/privileges/read-match-merge", "execute");
 
 const common = require("/data-hub/5/data-services/mastering/validateStepCommonLib.sjs");
 const step = require('/data-hub/5/artifacts/core.sjs').getArtifact("merging", stepName);
 
 let warnings = [];
-let targetCollections = Object.assign({onArchive:[], onNoMatch:[], onMerge:[], onNotification:[], onAuditing:[]}, step.targetCollections);
-let allCollections = []
-    .concat(step.collections)
-    .concat(step.additionalCollections)
-    .concat(targetCollections.onArchive)
-    .concat(targetCollections.onNoMatch)
-    .concat(targetCollections.onMerge)
-    .concat(targetCollections.onNotification)
-    .concat(targetCollections.onAuditing)
-if (allCollections.length) {
-    let targetTypeWarning = common.targetEntityCollectionWarning(step.targetEntityType, allCollections);
-    if (targetTypeWarning) {
-        warnings.push(targetTypeWarning);
+if (view === "settings") {
```

Comment:
If this code migrates out into a library, I think it'd be worth breaking this down into smaller functions for readability - i.e. validateSettings and validateRules. 

Also, I've been wondering if we should refactor at some point to having classes for steps - e.g. we'd have a MergingStep class, and we'd be adding new "validateSettings" and "validateRules" functions to it. The DS endpoint would then load the step from the database, pass the JSON into a new instance of MergingStep, and call either of those functions. Just something to think about - possibly a direction to head in during 5.6.